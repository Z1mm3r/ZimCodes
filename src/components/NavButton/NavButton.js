import React from 'react'
import Button from '@material-ui/core/Button'
import {Link, useRouteMatch } from 'react-router-dom' 


const NavButton = (props) => {

    const { path, label } = {...props}
    const match = useRouteMatch(path)
    const RenderButton = () => {
        return  match && match.isExact ? (
            <Button color = "secondary"  variant = "outlined">
                { label }
            </Button>
        )
        : (
            <Link data-testid="navButtonLink" to={path}>
                <Button color="primary" variant = "outlined">
                    { label }
                </Button>
            </Link>
        )
    }

    return(
        <>
            {RenderButton()}
        </>
    )
}

export default  NavButton