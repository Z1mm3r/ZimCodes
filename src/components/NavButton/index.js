import React from 'react'
import NavButton from './NavButton.js'


const index = (props) =>{
    return(
        <>
            <NavButton {...props}/>
        </>
    )
}

export default index