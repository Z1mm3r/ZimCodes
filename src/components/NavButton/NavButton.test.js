import NavButton from './index'
import React from 'react'
import {screen} from '@testing-library/react'

import {renderWithRouter} from 'utils/testUtils.js'
  
  
  test('renders without crashing', () => {
    const {asFragment} = renderWithRouter(<NavButton path='/'/>)
    expect(asFragment()).toMatchSnapshot();
  })

  test('when the route does not match, is a clickable Link',() =>{
      renderWithRouter(<NavButton path="/wrong"/>)
      const linkElement = screen.getByTestId('navButtonLink')
      expect(linkElement).toBeTruthy()

  })

  test('when the route does  match, there is no clickable Link',() =>{
      renderWithRouter(<NavButton path="/"/>)
      expect(() => (screen.getByTestId('navButtonLink'))).toThrow()
})