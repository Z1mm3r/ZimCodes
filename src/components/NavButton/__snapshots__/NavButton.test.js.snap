// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`when the route does not match, is a clickable Link 1`] = `
<DocumentFragment>
  <a
    data-testid="navButtonLink"
    href="/wrong"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </a>
</DocumentFragment>
`;
